docopt.cpp

```diff
@@ -17,11 +17,26 @@
 #include <unordered_map>
 #include <map>
 #include <string>
-#include <regex>
 #include <iostream>
 #include <cassert>
 #include <cstddef>
 
+// Workaround GCC 4.8 not having std::regex
+#if (__GNUC__ == 4) && (__GNUC_MINOR__ == 8)
```

Comment:
I am not sure what is the best thing to do here. We can either make the use of Boost.Regex opt-in (with some flag), or default like this.

We also can add some preprocessor warning to the user:
```cpp
#if (__GNUC__ == 4) && (__GNUC_MINOR__ == 8)
    #ifdef DOCTOPT_USE_BOOST_REGEX
        #include <boost/regex.hpp>
        [...]
    #else
        #error "GCC 4.8 can not build docopt alone. You can opt-in to using Boost.Regex by defining DOCTOPT_USE_BOOST_REGEX"
    #endif
#else 
#include <regex> 
#endif
```